### PR TITLE
Debian packaging improvements (module autoloading)

### PR DESCRIPTION
**Pull request by pulkomandy**
This merge request changes two things:

- Include the provided udev rules in the Debian package
- When installing the package, load the module, and make sure it is loaded at boot as well. Since this module does not correspond to actual hardware, it otherwise needs to be loaded manually.

### CODE REVIEW

**Comment by stappersg** [comment]
(Sorry for the late response)

The two commits in this merge request should be in at least two separate MRs.

**Comment by pulkomandy** [comment]
Hello,

Thanks for updating this tool :)

I created a separate merge request for just the udev rules: https://github.com/freemed/tty0tty/pull/61

**Comment by pulkomandy** [comment]
Now I have removed that part from this pull request, so you can merge them independently.

**Comment by stappersg** [comment]
On Mon, Sep 16, 2024 at 04:02:43AM -0700, Adrien Destugues wrote:
> 
> +DKMS_VERSION=1.2
> 
> This needs to be updated when/if a version 1.3 of tty0tty (orany
> other new version number) is released. I'm not sure if it is possible
> to extract this info from somewhere else,
> what would be considered appropriate here?

Appropriate would be discussioning it outside this merge request.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by stappersg** [comment]
For the record:
Some part of this merge request are good,  some parts not.

It is lack of resources to merge this request **and** to undo the not so good parts of the MR.

**Comment by pulkomandy** [comment]
Hello,

You requested that I split the merge request in two parts, and I already did that.

If you have other requests, please state them, because I cannot work from "there are some parts that are not good".

**Comment by stappersg** [comment]
On Mon, Sep 30, 2024 at 12:06:56AM -0700, Adrien Destugues wrote:
> Hello,
 
Hi there,


> You requested that I split the merge request in two parts,
> and I already did that.

Acknowledge

 
> If you have other requests, please state them,
> because I cannot work from "there are some parts that are not good".

Also true.


The good part: There is `postinst`.

The not so good parts:
- There is no `postrm`.
- The silent "load the kernel module on start up"
- Version number nowhere documented
- The unneeded modification of the .gitignore



The .gitignore modification that I deem unneeded, can I leave out.
If the ignore change is needed, put it in a separate merge request
with a commit message that explains the **why**.


The github comment thread of this "pull request" has already
tipped "this version numberhere, will effect releases". It waits
for a follow-up.


The "load the kernel module" is indeed the purpose of `tty0tty`.
Currently it feels wrong to me how it is done. I might wrong.
I think the what I deem "odd" can be overcome with debconf.
Most likely it is a non-issue.  Or an issue that can be addressed
in debian/README


Upon uninstall of the Debian package should the `postinst` be undone.


In a near future is none of the above on **my** priority list.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by stappersg** [comment]
For what is it worth:  #69   _created_ **tag** `1.3.0`.
